std/algorithm/iteration.d

```diff
@@ -4289,3 +4304,625 @@ unittest
          [1, 2, 0],
          [2, 1, 0]]));
 }
+
+/**
+Computes the $(WEB https://en.wikipedia.org/wiki/Binomial_coefficient, binomial coefficient)
+of $(D n,k).
+It is also known as "n choose k " or more formally as $(D n!/k!(n-k)).
+
+Params:
+    n = non negative integer
+    k = non negative integer, k <= n
+
+Returns:
+    binomial coefficient
+*/
+
+
+size_t binomial(size_t n, size_t k) pure nothrow @safe @nogc
```

Comment:
I don't know which of std.math and std.numeric is most appropriate. @9il @ibuclaw?